modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/SqlPartitionEvictionTest.java

```diff
@@ -0,0 +1,210 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.index;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cache.CacheWriteSynchronizationMode;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.query.SqlFieldsQuery;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.GridCacheContext;
+import org.apache.ignite.internal.processors.cache.persistence.CacheDataRow;
+import org.apache.ignite.internal.processors.query.GridQueryProcessor;
+import org.apache.ignite.internal.processors.query.GridQueryTypeDescriptor;
+import org.apache.ignite.internal.processors.query.h2.IgniteH2Indexing;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/** */
+@RunWith(Parameterized.class)
+public class SqlPartitionEvictionTest extends GridCommonAbstractTest {
+    /** */
+    private static final String POI_CACHE_NAME = "POI_CACHE";
+
+    /** */
+    private static final String POI_SCHEMA_NAME = "DOMAIN";
+
+    /** */
+    private static final String POI_TABLE_NAME = "POI";
+
+    /** */
+    private static final String POI_CLASS_NAME = "PointOfInterest";
+
+    /** */
+    private static final String ID_FIELD_NAME = "id";
+
+    /** */
+    private static final String NAME_FIELD_NAME = "name";
+
+    /** */
+    private static final String LATITUDE_FIELD_NAME = "latitude";
+
+    /** */
+    private static final String LONGITUDE_FIELD_NAME = "longitude";
+
+    /** */
+    private static final int NUM_ENTITIES = 1_000;
+
+    /** Test parameters. */
+    @Parameterized.Parameters(name = "backups_count={0}")
+    public static Iterable<Object[]> params() {
+        return Arrays.asList(
+                new Object[] { 0 },
+                new Object[] { 1 },
+                new Object[] { 2 }
+            );
+    }
+
+    /** */
+    @Parameterized.Parameter
+    public int backupsCount;
+
```

Comment:
Missing javadoc

Reply:
Latch is used in static context of BlockingIndexing

Reply:
Opps, sorry, I missed it. Ok, there is no variant except this one.